### PR TITLE
Exodus Intelligence Contribution

### PR DESCRIPTION
**Pull request by an-hel**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by JasBeilin** [comment]
Hi @an-hel , please fill the contribution form (https://github.com/demisto/content/pull/19383#issuecomment-1145594657) and sign the cla (https://github.com/demisto/content/pull/19383#issuecomment-1145594081) so we can start the review :)

**Comment by JasBeilin** [comment]
Hi @an-hel , Please fill the contribution form so we can start your review.

File: Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    """
+    if scores == []:
+        return 0.0
+    commented_scores = [s for s in scores if s["comment"] != ""]
+    if len(commented_scores) > 0:
+        scores = commented_scores
+    return float(sorted(scores, key=lambda x: x["created_at"], reverse=True)[0]["score"])
+
+
+def extract_data(data, dest):
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py:65]: I think this function can be replaced with a few demisto.get(data, key) calls.

File: Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                        report_data = extract_data(report, report_data)
+                    vuln_data = extract_data(vulnerability, report_data)
+                    formatted_list.append(vuln_data)
+            except KeyError as e:
+                demisto.debug(f"There was a problem: {e}")
+    except Exception as e:
+        demisto.debug(f"Something went wrong: {e}")
+        return False
+
+    if len(formatted_list):
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py:159]:
```suggestion
    if formatted_list:
```

File: Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+                indicator = {
+                    "value": items["identifier"],
+                    "type": "Exodus Intelligence",
+                    "fields": items,
+                }
+                indicators.append(indicator)
+            except KeyError as e:
+                demisto.debug(f"There was a problem creating indicators: {e}")
+        demisto.createIndicators(indicators)
+    return True
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py:171]: You are not using the return value so I think you can remove it.

File: Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+
+def reset_data_stream(days):
+    try:
+        exodus = connect()
+        data = exodus.get_recent_vulns(days)
+        return data["data"]["end_ts"]
+    except (ConnectionError, KeyError) as e:
+        demisto.debug(f"There was a problem connecting to the API {e}")
+        return "Unable to reset Exodus Intelligence data stream."
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py:181]: May I suggest raising the exception, and then catching it in the main when you handle the error? It should probably fail the command.

File: Packs/ExodusIntelligence/TestPlaybooks/Test.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      "dimensions": {
+        "height": 65,
+        "width": 380,
+        "x": 50,
+        "y": 50
+      }
+    }
+  }
+fromversion: 6.2.0
+description: ''
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/TestPlaybooks/Test.yml:47]: It seems to me this Test playbook is empty, is it on purpose? Please change it to test the commands :)

File: Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/test_data/baseintegration-dummy.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/test_data/baseintegration-dummy.json:1]: Please erase this file as it is empty.

File: Packs/ExodusIntelligence/README.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+**Exodus Intelligence’s unmatched vulnerability research capabilities and products provide unique context around your enterprise infrastructure allowing you to better prioritize protection and mitigation efforts. This rich context in conjunction with Cortex XSOAR enables many value-add use cases for alerting and contextual enrichment that would otherwise not be possible.**
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/README.md:1]: Please provide example for the commands and generate the readme again, as I believe it was missed.

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Thank you for your contribution!
Please see my comments below. Let me know if you need help.

File: Packs/ExodusIntelligence/TestPlaybooks/Test.yml (hunk 5 above)
Comment (an-hel) [reply Packs/ExodusIntelligence/TestPlaybooks/Test.yml:47]: The file was auto-generated by the SDK. We don't have any playbooks.

File: Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py (hunk 1 above)
Comment (JasBeilin) [reply Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py:56]: In both cases demisto.get() will do the job.

File: Packs/ExodusIntelligence/.secrets-ignore [hunk 8]
```diff
@@ -0,0 +1,2 @@
+https://vpx.exodusintel.com
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/.secrets-ignore:1]:
```suggestion
hxxps://vpx.exodusintel.com
```

Or use dummy urls please.

File: Packs/ExodusIntelligence/.secrets-ignore [hunk 9]
```diff
@@ -0,0 +1,2 @@
+hxxps://vpx.exodusintel.com
```
Comment (JasBeilin) [inline Packs/ExodusIntelligence/.secrets-ignore:1]: I now understand that the secret is for the actual URL you are providing to customer, I am sorry for misunderstanding. 

```suggestion
https://vpx.exodusintel.com
```

**Review by JasBeilin — APPROVED** [review]
Please commit my last comment's fix, and we can move forward and schedule a demo.

File: Packs/ExodusIntelligence/.secrets-ignore (hunk 9 above)
Comment (an-hel) [reply Packs/ExodusIntelligence/.secrets-ignore:1]: done

**Comment by JasBeilin** [comment]
We are waiting for an additional review from @idovandijk. We can schedule a demo in the mean time. Please notice you have some file that needs to be removed:
Packs/MISP/pack_metadata.json.lock
Packs/ExodusIntelligence/pack_metadata.json.lock

**Comment by Ni-Knight** [comment]
Good morning,
I don't see any layout attached to the PR, since youre creating a new indicator type with specific fields you should prob create a unique layout as well.

I would also think it will be better if you checked if the CVE is already an IOC in TIM and create a relationship with it and if not create it with basic data and create a relationship between it and your new type (I know our CVE type is limited, for now at least 🙂). This way it will be easier for customers to relate the data from exodus feed to IOCs they already have.

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-CPE23URI.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    "locked": false,
+    "name": "CPE23URI",
+    "neverSetAsRequired": false,
+    "openEnded": true,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "multiSelect",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-CPE23URI.json:24]: This should probably be a grid field if I understand your meaning correctly

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-CPE_MATCH.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    "locked": false,
+    "name": "CPE MATCH",
+    "neverSetAsRequired": false,
+    "openEnded": false,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "longText",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-CPE_MATCH.json:24]: if its longtext its better to use markdown as it gives you better options with no real downside.

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Links.json [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    "locked": false,
+    "name": "Links",
+    "neverSetAsRequired": false,
+    "openEnded": true,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "multiSelect",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Links.json:24]: Grid as well.

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Methods.json [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    "locked": false,
+    "name": "Methods",
+    "neverSetAsRequired": false,
+    "openEnded": true,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "multiSelect",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Methods.json:21]: Tags here \ Grid

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Mitigation.json [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    "locked": false,
+    "name": "Mitigation",
+    "neverSetAsRequired": false,
+    "openEnded": false,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "longText",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Mitigation.json:24]: markdown

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Modules.json [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    "locked": false,
+    "name": "Modules",
+    "neverSetAsRequired": false,
+    "openEnded": true,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "multiSelect",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Modules.json:24]: tags

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Ports.json [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+    "locked": false,
+    "name": "Ports",
+    "neverSetAsRequired": false,
+    "openEnded": true,
+    "ownerOnly": false,
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "multiSelect",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-Ports.json:24]: definitely tags

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-ZDIS.json [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    "cliName": "zdis",
+    "closeForm": false,
+    "content": true,
+    "editForm": true,
+    "group": 2,
+    "hidden": false,
+    "id": "indicator_zdis",
+    "isReadOnly": false,
+    "locked": false,
+    "name": "ZDIS",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-ZDIS.json:16]: what are ZDIs?

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-cpes.json [hunk 18]
```diff
@@ -0,0 +1,34 @@
+{
+    "id": "indicator_cpes",
+    "version": -1,
+    "modified": "2022-04-29T11:08:49.074455322-04:00",
+    "propagationLabels": [
+        "all"
+    ],
+    "name": "CPES",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-cpes.json:8]: Why are there 3 different fields for CPEs parts? what is going to be in this one?

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-product.json [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+    "id": "indicator_product",
+    "version": -1,
+    "modified": "2022-04-26T16:03:36.753192181-04:00",
+    "propagationLabels": [
+        "all"
+    ],
+    "name": "Product",
+    "ownerOnly": false,
+    "cliName": "product",
+    "type": "longText",
```
Comment (Ni-Knight) [inline Packs/ExodusIntelligence/IndicatorFields/indicatorfield-product.json:11]: Short text might be enough here, if not maybe markdown.

**Review by Ni-Knight — COMMENTED** [review]
We def need a demo as its hard to visualize it from just the code, looks good all in all but I do have some tips for creating an even better pack.

**Comment by an-hel** [comment]
> Hi @an-hel, are you having trouble with fixing the PR by our reviews? If so please let me know.

Hi @JasBeilin, I was out last week. I should repost something soon.

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-cpes.json (hunk 18 above)
Comment (an-hel) [reply Packs/ExodusIntelligence/IndicatorFields/indicatorfield-cpes.json:8]: What are the other parts you are referring to?

File: Packs/ExodusIntelligence/IndicatorFields/indicatorfield-CPE23URI.json (hunk 10 above)
Comment (an-hel) [reply Packs/ExodusIntelligence/IndicatorFields/indicatorfield-CPE23URI.json:24]: For some reason, the grid type doesn't get populated. Leaving it as it is until further investigation.

**Comment by an-hel** [comment]
The Google Form we filled out contains a link to the demo video requested. We can also schedule a video conference to demo it. Please let me know.

**Comment by an-hel** [comment]
@JasBeilin @Ni-Knight, any new updates on this?

**Comment by JasBeilin** [comment]
@an-hel , Sorry, I was not aware to the fact that you finished the changes. I will go over it as soon as I can.

**Comment by JasBeilin** [comment]
Looks good!
Please pull from master (in both your main branch and the contrib branch, so we can run all our checks and then I will approve :)
Also, lets schedule a demo so we can move forward.

**Comment by JasBeilin** [comment]
I can see your PR validation still fails on non-updated master. Please make sure to update both your contribution branch and your main branch.

**Comment by an-hel** [comment]
@JasBeilin, any news?

**Comment by JasBeilin** [comment]
Hi @an-hel Please remove the Packs/MISP/pack_metadata.json.lock file,  as it is not related to your contrib.
I am working on understanding why your build is failing and scheduling a demo so we can move foreward.

**Comment by an-hel** [comment]
> Hi @an-hel Please remove the Packs/MISP/pack_metadata.json.lock file, as it is not related to your contrib. I am working on understanding why your build is failing and scheduling a demo so we can move foreward.

Do you want me to delete that file even though I didn't touch that pack? It seems the issue is not with our build, but with the demisto SDK itself and some dependencies [Look](https://github.com/demisto/demisto-sdk/issues/2097)

**Comment by JasBeilin** [comment]
@an-hel yes, there are a few files that were changed (probably by mistake)
[package.json](https://github.com/demisto/content/pull/19383/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) 

and package-lock.json
https://github.com/demisto/content/pull/19383/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519

reverting this files (and the MISP file) and pulling from master should fix this build failures.

**Comment by JasBeilin** [comment]
Great! let's schedule a demo asap. Is there a preferred time for you?

**Comment by an-hel** [comment]
Ok, let's schedule the demo.
Availability for the week of August 14th - August 19th: 
Monday at 12p or after 3p
Tuesday at 12p
Wednesday after 3p
Thursday after 1p

**Comment by JasBeilin** [comment]
Updating here for followups- We discussed via slack on some issues with the fetching mechanism and scheduled a demo.

**Comment by JasBeilin** [comment]
Hi @an-hel, any updates? can I help you in any kind of way?

**Comment by JasBeilin** [comment]
Hi @an-hel, please merge from master so the build will start running, as it currently fails on it.

**Review by Ni-Knight — APPROVED** [review]

**Pull request merged**